### PR TITLE
fix: universal resolver http error

### PR DESCRIPTION
**Pull request by TateB**
previously, UR was throwing a generic revert error where `HttpError` should be thrown because the decoding method used was incorrect. this PR fixes that by just dumb reverting the entire result (which includes the `HttpError` selector)

changes:
- switched to directly propagate the entire `HttpError` without any decoding/encoding

### CODE REVIEW

File: test/utils/TestUniversalResolver.js [hunk 1]
```diff
@@ -560,6 +560,45 @@ contract('UniversalResolver', function (accounts) {
...
+        [true],
+        [errorData],
+      ])
+
+      try {
+        await urWithHttpErrorAbi.callStatic.resolveSingleCallback(
+          responses,
+          extraData,
+        )
+        expect(false).to.be.true
```
Comment (jefflau) [inline test/utils/TestUniversalResolver.js:592]: `expect(false).to.be.true` this seems weird - is this supposed to be like this?

Comment (TateB) [reply test/utils/TestUniversalResolver.js:592]: yes, we're using waffle chai matchers which are pretty outdated so afaik this is the only way to assert that a call has reverted if i want to access the data of the error

Comment (TateB) [reply test/utils/TestUniversalResolver.js:592]: there were actually a few missing so i added those as well

**Review by jefflau — APPROVED** [review]

**Comment by Mrt3383x38** [comment]
Binance eror lütfen

**Pull request merged**